### PR TITLE
first commit 

### PR DESCRIPTION
**Pull request by anhhuy1010**

### CODE REVIEW

File: controllers/product.go [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+	cond := bson.M{}
+	if req.Name != nil {
+		cond["name"] = req.Name
+	}
+	if req.IsActive != nil {
+		cond["is_active"] = req.IsActive
+	}
+
+	if req.MaxQuantity != nil && req.MinQuantity != nil {
+		cond["quantity"] = bson.M{"$gt": req.MinQuantity, "$lt": req.MaxQuantity}
```
Comment (namleDiqit) [inline controllers/product.go]: Should be $gte and $lte

File: controllers/product.go [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+	} else if req.MinQuantity != nil {
+		cond["quantity"] = bson.M{"$lt": req.MinQuantity}
+	} else if req.MaxQuantity != nil {
+		cond["quantity"] = bson.M{"$gt": req.MaxQuantity}
+	}
+	if req.StartDate != nil {
+		cond["start_date"] = bson.M{"$gte": req.StartDate}
```
Comment (namleDiqit) [inline controllers/product.go]: Should be req.AvailableDate

File: controllers/product.go [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+	}
+	productt.IsActive = *req.IsActive
+
+	_, err = productt.Update()
+	if err != nil {
+		_ = c.Error(err)
+		c.JSON(http.StatusOK, respond.UpdatedFail())
+		return
+	}
+	c.JSON(http.StatusOK, respond.Success(productt.IsActive, "update successfully"))
```
Comment (namleDiqit) [inline controllers/product.go]: Should be product.Uuid

File: controllers/product.go [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+		return
+	}
+	c.JSON(http.StatusOK, respond.Success(productt.Uuid, "Delete successfully"))
+}
+
+// Function Create
+func (productClt ProductController) Create(c *gin.Context) {
+	var req request.GetInsertRequest
+	err := c.ShouldBindWith(&req, binding.Query)
+	if err != nil {
```
Comment (namleDiqit) [inline controllers/product.go]: not use c.ShouldBindWith(&req, binding.Query) in Create func

File: controllers/product.go [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+		c.JSON(http.StatusOK, respond.ErrorCommon("Product no found!"))
+		return
+	}
+	if req.IsActive != nil {
+		productt.IsActive = *req.IsActive
+	}
+	if req.Name != "" {
+		productt.Name = req.Name
+	}
+	if req.Image != "" {
```
Comment (namleDiqit) [inline controllers/product.go]: Field Image, Description, Price, Sequence, Quantity can be optional, can set with default value.
Missing Gallery field

File: grpc/proto/user/user.pb.go
Comment (namleDiqit) [inline grpc/proto/user/user.pb.go:1]: Revert this file

File: grpc/service/user.go
Comment (namleDiqit) [inline grpc/service/user.go:1]: Revert this file

File: request/products/insertdata.go
Comment (namleDiqit) [inline request/products/insertdata.go:1]: Rename this file

File: request/products/updatestatus.go
Comment (namleDiqit) [inline request/products/updatestatus.go:1]: Rename this file

**Review by LeHNam — APPROVED** [review]

**Pull request merged**